src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -890,22 +900,27 @@ void VideoDecoder::convertAVFrameToDecodedOutputOnCPU(
   if (output.streamType == AVMEDIA_TYPE_VIDEO) {
     if (streamInfo.colorConversionLibrary == ColorConversionLibrary::SWSCALE) {
       torch::Tensor tensor;
+      int width = streamInfo.options.width.value_or(frame->width);
+      int height = streamInfo.options.height.value_or(frame->height);
       if (preAllocatedOutputTensor.has_value()) {
-        // TODO: check shape of preAllocatedOutputTensor?
         tensor = preAllocatedOutputTensor.value();
+        auto shape = tensor.sizes();
+        TORCH_CHECK(
+            (shape.size() == 3) && (shape[0] == height) &&
+                (shape[1] == width) && (shape[2] == 3),
+            "Expected tensor of shape ",
+            height,
+            "x",
+            width,
+            "x3, got ",
+            shape);
```

Comment:
Any idea how to make this single call shorter :thinking: ?

Reply:
I think you can do:
```
TORCH_CHECK(
            (shape.size() == 3) && (shape.equals({height, width, 3}),
            "Expected tensor of shape ",
            height,
            "x",
            width,
            "x3, got ",
            shape);
```
But I'm not sure. The main thing I'm not sure about is if an array literal will auto-convert into a the corresponding `ArrayRef`: https://pytorch.org/cppdocs/api/classc10_1_1_array_ref.html#_CPPv4NK3c108ArrayRef6equalsE8ArrayRef

Reply:
Ah, I was mainly hoping to avoid the 

```
            height,
            "x",
            width,
            "x3, got ",
            shape);
```

stack :p